randovania/patching/prime/all_prime_dol_patches.py

```diff
@@ -58,8 +58,29 @@ class BasePrimeDolVersion(DolVersion):
 _remote_execution_max_byte_count = 420  # Prime 1 0-00 is 444, Echoes NTSC is 464
 
 
-def remote_execution_patch_start() -> list[BaseInstruction]:
+def remote_execution_patch_start(game: RandovaniaGame) -> list[BaseInstruction]:
     return_code = remote_execution_cleanup_and_return()
+    return_code_byte_count = assembler.byte_count(return_code)
+
+    if game == RandovaniaGame.METROID_PRIME:
+        cutscene_check = [
+            lwz(r3, 0x870, r31),
+            cmpwi(r3, 0),
+            beq(0x10, True),
+            lwz(r0, 0x8, r3),
+            cmpwi(r0, 0),
+            ble(return_code_byte_count + 4, True),
+            *return_code,
```

Comment:
The cutscene check is being added right after `intro`, which ends in `return_code`. We can reuse that same return block instead of duplicating it here.
In particular, the cutscene check should just be added to `intro`.

Reply:
so I put *return_code after *cutscene_check?